circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -56,13 +100,52 @@ public void removeTableParamsAndDrop(
       }
       client.alter_table(databaseName, tableName, table);
     }
-    LOG
-        .info("Dropping table '{}.{}'.", table.getDbName(), table.getTableName());
-    client.dropTable(table.getDbName(), table.getTableName(), false, true);
+    LOG.info("Dropping table '{}.{}'.", databaseName, tableName);
+    client.dropTable(databaseName, tableName, false, true);
+  }
+
+  private void deleteData(DataManipulationClient dataManipulationClient, String replicaDataLocation) {
+    try {
+      LOG.info("Dropping table data from location: {}.", replicaDataLocation);
+      boolean dataDeleted = dataManipulationClient.delete(replicaDataLocation);
+      LOG.info("Data deleted: {}.", dataDeleted);
+    } catch (IOException e) {
+      LOG.info("Could not drop replica table data at location:{}.", replicaDataLocation);
+    }
+  }
+
+  private void deletePartitionData(
+      DataManipulationClient dataManipulationClient,
+      String replicaTableLocation,
+      List<String> replicaPartitionLocations) {
+    try {
+      LOG.info("Dropping partition data from base location: {}.", replicaTableLocation);
+      for (String location : replicaPartitionLocations) {
+        boolean deleted = dataManipulationClient.delete(location);
+        LOG.debug("Attempted to delete data from location: {}. Successful deletion = {}.", location, deleted);
+      }
+    } catch (IOException e) {
```

Comment:
At the moment this will stop if one of the partitions couldn't be dropped. Wouldn't it be better to put the try/catch around line 124 so it can try each location? And then basically do an AND on the booleans returned so if any one is false then the method returns false.

Reply:
I think we should log on line 124 before doing the delete which location we're trying to delete and then let the exception bubble up. Why try all of them if 1 fails and 99 succeeds shouldn't we stop the replication?

Reply:
OK, as long as the approach is consistent, I'm fine with it being "as soon as anything goes wrong, ABORT AND FAIL!"